### DISCUSSION TITLE
Toggle key

### DISCUSSION THREAD

**Discussion by hardbrocklife** [sezanzeb/input-remapper#838] [Q&A]
I might just be having a massive brain fart, but I cant figure out how I would make a toggle-able key via the documentation(I also have a spinal nerve compression that is causing a 9/10 on the pain scale hampering problem solving).

All Im trying to figure out how to do is a Press KEY once = KEY depressed > Press KEY again = KEY released.

Something like this that could be assigned to any key.

I did search first and it looks like the questions about toggling were about toggling Input Remapper itself on and off.

**Comment by michaelBelsanti** [comment]
I'm also trying to do this. It doesn't seem to be possible at the moment.

**Comment by blamarche** [comment]
I really wish this program had this feature. It'd be so great for accessibility

**Comment by tomasmark79** [comment]
It is easy. I am using my own keybord layoute and I am using Right Alt like a toggle.
```
if_eq($toggle_state, 0,
    set(toggle_state, 1).key_down(ISO_Level3_Shift),
    set(toggle_state, 0).key_up(ISO_Level3_Shift))
```

No need explicitely set the variable at the begining of macro. It begin exist with first set command within the condition.

**Reply by TherinS** [reply]
Could you break this down, detailing what each part of it does and what I am supposed to put in each area?  I copy/pasted this into a profile and it didn't do anything that I could tell.  I am guessing that "$toggle_state" is supposed to be something specific, and the "ISO...." parts are supposed to be the key to be toggled?  I very much want this for my mouse button for actions like chopping trees in a game, where each button press swings the axe.  

Currently, I have a macro for starting the swinging with one button but had to map a 'stop swinging' on another button.  It would be nice to have a single button to start and stop the action.
